rcl/src/rcl/validate_security_context_name.c

```diff
@@ -0,0 +1,148 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rcl/validate_security_context_name.h"
+
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <rcutils/snprintf.h>
+
+#include "rmw/validate_namespace.h"
+
+#include "rcl/error_handling.h"
+
+#include "./common.h"
+
+rcl_ret_t
+rcl_validate_security_context_name(
+  const char * security_context,
+  int * validation_result,
+  size_t * invalid_index)
+{
+  if (!security_context) {
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+  return rmw_validate_namespace_with_size(
+    security_context, strlen(security_context), validation_result, invalid_index);
+}
+
+rcl_ret_t
+rcl_validate_security_context_name_with_size(
+  const char * security_context,
+  size_t security_context_length,
+  int * validation_result,
+  size_t * invalid_index)
+{
+  if (!security_context) {
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+  if (!validation_result) {
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+
+  int t_validation_result;
+  size_t t_invalid_index;
+  rmw_ret_t ret = rmw_validate_namespace_with_size(
+    security_context, security_context_length, &t_validation_result, &t_invalid_index);
+  if (ret != RMW_RET_OK) {
+    return rcl_convert_rmw_ret_to_rcl_ret(ret);
+  }
+
+  if (t_validation_result != RMW_NAMESPACE_VALID &&
+    t_validation_result != RMW_NAMESPACE_INVALID_TOO_LONG)
+  {
+    switch (t_validation_result) {
+      case RMW_NAMESPACE_INVALID_IS_EMPTY_STRING:
+        *validation_result = RCL_CONTEXT_NAME_INVALID_IS_EMPTY_STRING;
+        break;
+      case RMW_NAMESPACE_INVALID_NOT_ABSOLUTE:
+        *validation_result = RCL_CONTEXT_NAME_INVALID_NOT_ABSOLUTE;
+        break;
+      case RMW_NAMESPACE_INVALID_ENDS_WITH_FORWARD_SLASH:
+        *validation_result = RCL_CONTEXT_NAME_INVALID_ENDS_WITH_FORWARD_SLASH;
+        break;
+      case RMW_NAMESPACE_INVALID_CONTAINS_UNALLOWED_CHARACTERS:
+        *validation_result = RCL_CONTEXT_NAME_INVALID_CONTAINS_UNALLOWED_CHARACTERS;
+        break;
+      case RMW_NAMESPACE_INVALID_CONTAINS_REPEATED_FORWARD_SLASH:
+        *validation_result = RCL_CONTEXT_NAME_INVALID_CONTAINS_REPEATED_FORWARD_SLASH;
+        break;
+      case RMW_NAMESPACE_INVALID_NAME_TOKEN_STARTS_WITH_NUMBER:
+        *validation_result = RCL_CONTEXT_NAME_INVALID_NAME_TOKEN_STARTS_WITH_NUMBER;
+        break;
+      default:
+        {
+          char default_err_msg[256];
+          // explicitly not taking return value which is number of bytes written
+          int ret = rcutils_snprintf(
+            default_err_msg, sizeof(default_err_msg),
+            "rcl_validate_security_context_name_with_size(): "
+            "unknown rmw_validate_namespace_with_size() result '%d'",
+            t_validation_result);
+          if (ret < 0) {
+            RCL_SET_ERROR_MSG(
+              "rcl_validate_security_context_name_with_size(): "
+              "rcutils_snprintf() failed");
```

Comment:
@ivanpauno it would be good to leave a trace of the original error, e.g. 
```suggestion
              "rcutils_snprintf() failed while reporting an unknown validation result");
```

Reply:
See https://github.com/ros2/rcl/pull/515/commits/e3d5ebdf6094274f7f74d752ecce3fbadec84150.